test/parameter_test.py

```diff
@@ -147,6 +153,15 @@ class NoopTask(luigi.Task):
     pass
 
 
+class PEP593Task(luigi.Task):
+    param: Annotated[int, luigi.IntParameter()]
```

Comment:
What happens if a user specifies `Annotated[bool, luigi.IntParameter()]`? It seems we will just ignore it. Is that right?

Reply:
I meant accidentally using `bool`  with `luigi.IntParameter`.

Reply:
Yeah sorry mistyped above. Type checkers will only care about the type annotation but Luigi/runtime will only care about the Luigi parameter. So in your example that would be an integer at runtime, meaning the run time behavior would be exactly the same as it is now. And because bool is a subtype of int there isn't any expression the type checker would flag. Now if the user did `Annotated[str, IntegerParameter]` the value would once again be an integer at runtime but their type checker would be very upset when they try to do `self.parameter / 1`.

Reply:
Thank you for the explanation @adriangb. I'm concerned about not checking the type earlier and only having errors when trying to use the parameter.

Should we check earlier to enforce types if they're being annotated, concerned that if we check later when things are actually used it will difficult the debug process for the developer?